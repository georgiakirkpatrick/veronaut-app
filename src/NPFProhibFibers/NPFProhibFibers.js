import React from 'react'
import formData from '../FORM_DATA'
import FormPage from '../FormPage/FormPage'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'

const noneOption = [
    {
        text: 'None of these',
        id: 100,
        checked: false
    }
]

const NPFProhibFibers = props => {
    const {
        currentPage,
        setPage,
        none,
        noneChange,
        pFiberChecks,
        pFiberChange,
        pageTurns
    } = props

    const nextButton = () => {
        const prohibFibs = []

        Object.keys(pFiberChecks).forEach(key => {
            if (pFiberChecks[key]) {
                prohibFibs.push(key)
            }
        })

        if (prohibFibs.length === 0 && !none[100]) {
            alert('Please select an option.')
        } else {
            setPage(currentPage + pageTurns)
        }
    }

    return (
        <div>
            <FormPage title='Fibers'>
                <FormPromptWithSub
                    prompt="Does the product's fabrics(s) comprise of 10% or more of any of following materials?  Exclude notions such as buttons, zippers, or embroidery thread."
                    promptSubtitle="Check all that apply and click 'next'."
                />

                <FormCheckboxSection 
                    options={formData.prohibitedFibers}
                    selectedOptions={pFiberChecks}
                    handleChange={pFiberChange}
                />

                <FormCheckboxSection  
                    options={noneOption}
                    selectedOptions={none}
                    handleChange={noneChange}
                />
            </FormPage>

            <NPFFooter
                buttons='prevNext'
                previousButton={() => setPage(currentPage - pageTurns)} 
                nextButton={() => nextButton()}
            />
        </div>
    )
}

NPFProhibFibers.defaultProps = {
    currentPage: 1,
    handleChange: () => {},
    none: {},
    noneChange: () => {},
    pageTurns: 1,
    pFiberChange: () => {},
    pFiberChecks: {},
    selectedOptions: [],
    setPage: () => {},
    setPFiberChecks: () => {}
}

export default NPFProhibFibers