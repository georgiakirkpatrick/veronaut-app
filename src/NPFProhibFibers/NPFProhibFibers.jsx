import React from 'react'
import formData from '../FORM_DATA'
import FormPage from '../FormPage/FormPage'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'

const noneNotSure = [
  {
    text: 'None of these',
    id: 100,
    checked: false
  }
]

const options = formData.prohibitedFibers.slice(1, formData.prohibitedFibers.length)

const formattedProhib = options.sort((a, b) => 
  a.text > b.text ? 1 : -1
)

const prohibOptions = [
  ...formattedProhib,
  {
    text: 'Not sure',
    id: 101,
    checked: false
  }
] 

const NPFProhibFibers = props => {
  const {
    currentPage,
    setPage,
    none,
    noneChange,
    pFiberChecks,
    pFiberChange,
    pageTurns
  } = props

  const nextButton = () => {
    const prohibFibs = []

    Object.keys(pFiberChecks).forEach(key => {
      if (pFiberChecks[key]) {
        prohibFibs.push(key)
      }
    })

    if (prohibFibs.length === 0 && !none[100]) {
      alert('Please select an option.')
    } else {
      setPage(currentPage + pageTurns)
    }
  }

  return (
    <div>
      <FormPage title='Fibers'>
        <FormPromptWithSub
          prompt="Does the product's fabrics(s) comprise of 10% or more of any of following materials?  Exclude notions such as buttons, zippers, or embroidery thread."
          promptSubtitle="Check all that apply and click 'next'."
        />

        <FormCheckboxSection 
          options={prohibOptions}
          selectedOptions={pFiberChecks}
          handleChange={pFiberChange}
        />

        <FormCheckboxSection  
          options={noneNotSure}
          selectedOptions={none}
          handleChange={noneChange}
        />
      </FormPage>

      <NPFFooter
        buttons='prevNext'
        previousButton={() => setPage(currentPage - pageTurns)} 
        nextButton={() => nextButton()}
      />
    </div>
  )
}

NPFProhibFibers.defaultProps = {
  currentPage: 1,
  handleChange: () => {},
  none: {},
  noneChange: () => {},
  pageTurns: 1,
  pFiberChange: () => {},
  pFiberChecks: {},
  selectedOptions: [],
  setPage: () => {},
  setPFiberChecks: () => {}
}

export default NPFProhibFibers