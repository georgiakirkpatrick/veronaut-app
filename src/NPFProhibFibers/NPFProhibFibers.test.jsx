import React from 'react'
import { render, screen } from '@testing-library/react'
import NPFProhibFibers from './NPFProhibFibers'
import { describe, it, expect } from 'vitest'

describe ('NPFProhibFibers', () => {
  it ('renders without crashing', () => {
    render(<NPFProhibFibers />)
    screen.debug()
  })

  it('renders the UI as expected', () => {
    const tree = render(<NPFProhibFibers />)
    expect(tree).toMatchSnapshot()
  })
})