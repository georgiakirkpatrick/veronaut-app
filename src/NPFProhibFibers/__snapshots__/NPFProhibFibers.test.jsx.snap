// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NPFProhibFibers > renders the UI as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <section
          class="FormPage"
        >
          <h1>
            Fibers
          </h1>
          <form
            autocomplete="off"
            id="form-page"
          >
            <div
              class="FormPromptWithSub"
            >
              <p
                class="FormPromptWithSub__prompt"
              >
                Does the product's fabrics(s) comprise of 10% or more of any of following materials?  Exclude notions such as buttons, zippers, or embroidery thread.
              </p>
              <p
                class="FormPromptWithSub__subtitle"
              >
                Check all that apply and click 'next'.
              </p>
            </div>
            <div
              class="FormCheckboxSection"
            >
              <div
                class="empty"
              />
              <div
                class="FormCheckboxSection__options-container"
              >
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="1"
                    name="1"
                    type="checkbox"
                  />
                  <label
                    for="1"
                  >
                    Acetate
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="2"
                    name="2"
                    type="checkbox"
                  />
                  <label
                    for="2"
                  >
                    Acrylic
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="11"
                    name="11"
                    type="checkbox"
                  />
                  <label
                    for="11"
                  >
                    Elastane
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="13"
                    name="13"
                    type="checkbox"
                  />
                  <label
                    for="13"
                  >
                    Lycra
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="3"
                    name="3"
                    type="checkbox"
                  />
                  <label
                    for="3"
                  >
                    Microfiber
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="4"
                    name="4"
                    type="checkbox"
                  />
                  <label
                    for="4"
                  >
                    Neoprene
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="5"
                    name="5"
                    type="checkbox"
                  />
                  <label
                    for="5"
                  >
                    Nylon
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="6"
                    name="6"
                    type="checkbox"
                  />
                  <label
                    for="6"
                  >
                    Olefin
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="7"
                    name="7"
                    type="checkbox"
                  />
                  <label
                    for="7"
                  >
                    Polyamide
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="8"
                    name="8"
                    type="checkbox"
                  />
                  <label
                    for="8"
                  >
                    Polyester
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="9"
                    name="9"
                    type="checkbox"
                  />
                  <label
                    for="9"
                  >
                    Polyurethane
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="12"
                    name="12"
                    type="checkbox"
                  />
                  <label
                    for="12"
                  >
                    Spandex
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="10"
                    name="10"
                    type="checkbox"
                  />
                  <label
                    for="10"
                  >
                    Vinyl
                  </label>
                </div>
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="101"
                    name="101"
                    type="checkbox"
                  />
                  <label
                    for="101"
                  >
                    Not sure
                  </label>
                </div>
              </div>
            </div>
            <div
              class="FormCheckboxSection"
            >
              <div
                class="empty"
              />
              <div
                class="FormCheckboxSection__options-container"
              >
                <div
                  class="FormCheckboxSection__option"
                >
                  <input
                    id="100"
                    name="100"
                    type="checkbox"
                  />
                  <label
                    for="100"
                  >
                    None of these
                  </label>
                </div>
              </div>
            </div>
          </form>
          <div />
        </section>
        <footer
          class="NPFFooter"
        >
          <button
            class="NPFFooter__previous"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-left NPFFooter__left-arrow"
              data-icon="angle-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                fill="currentColor"
              />
            </svg>
            <span>
              PREVIOUS
            </span>
          </button>
          <button
            class="NPFFooter__next"
            type="button"
          >
            <span>
              NEXT
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angle-right NPFFooter__right-arrow"
              data-icon="angle-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
                fill="currentColor"
              />
            </svg>
          </button>
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <section
        class="FormPage"
      >
        <h1>
          Fibers
        </h1>
        <form
          autocomplete="off"
          id="form-page"
        >
          <div
            class="FormPromptWithSub"
          >
            <p
              class="FormPromptWithSub__prompt"
            >
              Does the product's fabrics(s) comprise of 10% or more of any of following materials?  Exclude notions such as buttons, zippers, or embroidery thread.
            </p>
            <p
              class="FormPromptWithSub__subtitle"
            >
              Check all that apply and click 'next'.
            </p>
          </div>
          <div
            class="FormCheckboxSection"
          >
            <div
              class="empty"
            />
            <div
              class="FormCheckboxSection__options-container"
            >
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="1"
                  name="1"
                  type="checkbox"
                />
                <label
                  for="1"
                >
                  Acetate
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="2"
                  name="2"
                  type="checkbox"
                />
                <label
                  for="2"
                >
                  Acrylic
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="11"
                  name="11"
                  type="checkbox"
                />
                <label
                  for="11"
                >
                  Elastane
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="13"
                  name="13"
                  type="checkbox"
                />
                <label
                  for="13"
                >
                  Lycra
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="3"
                  name="3"
                  type="checkbox"
                />
                <label
                  for="3"
                >
                  Microfiber
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="4"
                  name="4"
                  type="checkbox"
                />
                <label
                  for="4"
                >
                  Neoprene
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="5"
                  name="5"
                  type="checkbox"
                />
                <label
                  for="5"
                >
                  Nylon
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="6"
                  name="6"
                  type="checkbox"
                />
                <label
                  for="6"
                >
                  Olefin
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="7"
                  name="7"
                  type="checkbox"
                />
                <label
                  for="7"
                >
                  Polyamide
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="8"
                  name="8"
                  type="checkbox"
                />
                <label
                  for="8"
                >
                  Polyester
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="9"
                  name="9"
                  type="checkbox"
                />
                <label
                  for="9"
                >
                  Polyurethane
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="12"
                  name="12"
                  type="checkbox"
                />
                <label
                  for="12"
                >
                  Spandex
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="10"
                  name="10"
                  type="checkbox"
                />
                <label
                  for="10"
                >
                  Vinyl
                </label>
              </div>
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="101"
                  name="101"
                  type="checkbox"
                />
                <label
                  for="101"
                >
                  Not sure
                </label>
              </div>
            </div>
          </div>
          <div
            class="FormCheckboxSection"
          >
            <div
              class="empty"
            />
            <div
              class="FormCheckboxSection__options-container"
            >
              <div
                class="FormCheckboxSection__option"
              >
                <input
                  id="100"
                  name="100"
                  type="checkbox"
                />
                <label
                  for="100"
                >
                  None of these
                </label>
              </div>
            </div>
          </div>
        </form>
        <div />
      </section>
      <footer
        class="NPFFooter"
      >
        <button
          class="NPFFooter__previous"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-left NPFFooter__left-arrow"
            data-icon="angle-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              fill="currentColor"
            />
          </svg>
          <span>
            PREVIOUS
          </span>
        </button>
        <button
          class="NPFFooter__next"
          type="button"
        >
          <span>
            NEXT
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-right NPFFooter__right-arrow"
            data-icon="angle-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M278.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-160 160c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L210.7 256 73.4 118.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l160 160z"
              fill="currentColor"
            />
          </svg>
        </button>
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
